console_backend/src/tabs/main_tab.rs

```diff
@@ -150,7 +139,12 @@ impl MainTab {
         start_recording(&self.client_sender, filepath.display().to_string());
     }
 
-    pub fn serialize_frame(&mut self, frame: &Frame) {
+    /// Serialize frame incoming data,
+    ///
+    /// # Parameters:
+    /// - `frame`: The raw incoming data frame
+    /// - `msg`: Parsed message if present
+    pub fn serialize(&mut self, frame: &Frame, msg: Option<&Sbp>) {
```

Comment:
probably better to just have one or the other parameter but since we want to keep a copy of the computed frame -> msg to prevent recalculation, just keep both params

Reply:
also NOTE: this isn't just serialize, it also updates the sbp formats which might be nice to seperate into a different method and have this only do serialization stuff